src/GafferOSL/OSLObject.cpp

```diff
@@ -169,6 +225,7 @@ void OSLObject::hashProcessedObject( const ScenePath &path, const Gaffer::Contex
 
 	interpolationPlug()->hash( h );
 	h.append( inPlug()->fullTransformHash( path ) );
+	h.append( resampledInPlug()->fullAttributesHash( path ) );
```

Comment:
I think we've got our wires crossed somewhere - in `computeProcessedObject()` we call `resampledInPlug()->objectPlug()->getValue()`, so here we need to call `resampledInPlug()->objectPlug()->hash( h )`, and _not_ anything to do with the attributes.

Recap :

- A ScenePlug has various child plugs each representing a property of a location
    - `objectPlug()` represents geometry/cameras etc
    - `transformPlug()` represents the local transform
    - `boundPlug()` represents the local bounding box
    - `attributesPlug()` represents the local attributes. Note that "attributes" here is used in the RenderMan sense of inheritable properties that are orthogonal to geometry/transforms etc. This is different to the "everything is an attribute" approach taken by certain other packages.
- The `fullAttributes/fullAttributesHash` methods walk all parent locations, getting the local attributes (or their hash) and combining them with the result so far. They're rarely needed in a computation context.